### PR TITLE
Fix switch indentation for list patterns

### LINKED ISSUE #72196: List pattern is not correctly formatted in the first case of a switch expression
**Version Used**: '4.10.0-1.24081.13 (fa72fa61)' - Visual Studio 17.10.0 Preview 1.0

Language version: 12.0

**Steps to Reproduce**:

Format the following document in Visual Studio:

```c#
_ = Array.Empty<string>() switch
{
    [] => 0,
    _ => 1,
};
````

The list pattern is push to the left:

````c#
_ = Array.Empty<string>() switch
{
[] => 0,
    _ => 1,
};
````

It seems specific to the list pattern. Other patterns are correctly aligned.

**Expected Behavior**:

The list pattern stays at the same location.

**Actual Behavior**:

The list pattern is pushed to the left

### PR DESCRIPTION
**Pull request by obonn1**
fixes #72196

The indentation of a list pattern or collection expression is (according to pre-existing comment in the adjusted method) "relative to the start of the line the prior token is on if we're on the same line as it." However, the code was missing a check for the two tokens being on the same line.

### CODE REVIEW

File: src/Workspaces/CSharpTest/Formatting/FormattingTests.cs [hunk 1]
```diff
@@ -5605,6 +5605,34 @@ public object Method(int i)
             await AssertFormatAsync(expectedCode, code);
         }
 
+        [Fact, WorkItem("https://github.com/dotnet/roslyn/issues/72196")]
+        public async Task FormatSwitchExpression_ListPatternAligned()
+        {
```
Comment (CyrusNajmabadi) [inline src/Workspaces/CSharpTest/Formatting/FormattingTests.cs:5610]: can you please use 'raw string literals' for test cases.  There is a code refactoring to fix this up btw.

File: src/Workspaces/SharedUtilitiesAndExtensions/Compiler/CSharp/Formatting/Rules/IndentBlockFormattingRule.cs [hunk 2]
```diff
@@ -262,6 +262,11 @@ private static void AddBracketIndentationOperation(List<IndentBlockOperation> li
             // Brackets in list patterns are formatted like blocks, so align close bracket with open bracket
             AddIndentBlockOperation(list, bracketPair.openBracket.GetNextToken(includeZeroWidth: true), bracketPair.closeBracket.GetPreviousToken(includeZeroWidth: true));
 
+            if (IsSwitchExpressionPattern(node))
```
Comment (CyrusNajmabadi) [inline src/Workspaces/SharedUtilitiesAndExtensions/Compiler/CSharp/Formatting/Rules/IndentBlockFormattingRule.cs:265]: this needs a 'why' explanation as to why we don't want to execute teh rest of the code if we hit such a thing.

File: src/Workspaces/SharedUtilitiesAndExtensions/Compiler/CSharp/Formatting/Rules/IndentBlockFormattingRule.cs [hunk 3]
```diff
@@ -279,6 +284,11 @@ private static void AddBracketIndentationOperation(List<IndentBlockOperation> li
         }
     }
 
+    private static bool IsSwitchExpressionPattern(SyntaxNode node)
```
Comment (CyrusNajmabadi) [inline src/Workspaces/SharedUtilitiesAndExtensions/Compiler/CSharp/Formatting/Rules/IndentBlockFormattingRule.cs:287]: nit: i would *personally* inline this.  It's only called in one place, and has very simple logic.  If you do want to keep this as a helper method, that's ok. 

--

"nits" are ideas/gentle-suggestions.  But i'm deferring to your judgement here on what you prefer.

**Comment by obonn1** [comment]
@CyrusNajmabadi This should be ready for a new look whenever you have time

**Review by CyrusNajmabadi — APPROVED** [review]
Thanks!

**Pull request merged**